src/xPDO/xPDOContainer.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * This file is part of the xPDO package.
+ *
+ * Copyright (c) Jason Coward <jason@opengeek.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace xPDO;
+
+
+use Interop\Container\ContainerInterface;
+use xPDO\Exception\Container\NotFoundException;
+
+/**
+ * Implements a minimal container for service location.
+ *
+ * @package xPDO
+ */
+class xPDOContainer implements ContainerInterface, \ArrayAccess
+{
+    private $entries = [];
+
+    /**
+     * Add an entry to the container with the specified identifier.
+     *
+     * @param string $id The identifier for the entry.
+     * @param mixed  $entry The entry to add.
+     */
+    public function add($id, $entry)
+    {
+        $this->offsetSet($id, $entry);
+    }
+
+    /**
+     * Finds an entry of the container by its identifier and returns it.
+     *
+     * @param string $id Identifier of the entry to look for.
+     *
+     * @throws \Interop\Container\Exception\NotFoundException  No entry was found for this identifier.
+     * @throws \Interop\Container\Exception\ContainerException Error while retrieving the entry.
```

Comment:
It doesn't, this is just the signature from ContainerInterface. But perhaps I should catch any exceptions that might possibly be thrown and rethrow them as ContainerException.